test/integration/kubernetes_deploy_test.rb

```diff
@@ -667,24 +714,26 @@ def test_deploy_result_logging_for_mixed_result_deploy
     refute_logs_match("sidecar must exit 0") # this container is ready
 
     # Debug info for missing volume timeout
-    assert_logs_match_all([
+    expect_log = [
       %r{Deployment/missing-volumes: TIMED OUT \(progress deadline: \d+s\)},
       "Timeout reason: ProgressDeadlineExceeded",
       /Latest ReplicaSet: missing-volumes-\w+/,
       "Final status: 1 replica, 1 updatedReplica, 1 unavailableReplica",
-      /FailedMount.*secrets? "catphotoscom" not found/, # event
-    ], in_order: true)
+    ]
+    expect_log << /FailedMount.*secrets? "catphotoscom" not found/ unless server_dry_run_available?
```

Comment:
why does server_dry_run being available change the output of this tests?

Reply:
this output is from event, fetched by resource watcher. Events happen after template validation, so if server-dry-run catches the error during template validation. We won't have the event log